packages/babel-plugin-transform-es2015-parameters/src/default.js

```diff
@@ -44,17 +50,33 @@ const iifeVisitor = {
 };
 
 export const visitor = {
-  Function(path) {
+  Function(path, state) {
     const { node, scope } = path;
     if (!hasDefaults(node)) return;
 
     // ensure it's a block, useful for arrow functions
     path.ensureBlock();
 
-    const state = {
-      iife: false,
-      scope: scope,
-    };
+    const params = path.get("params");
+
+    if (state.opts.loose) {
+      const body = [];
+      for (let i = 0; i < params.length; ++i) {
+        const param = params[i];
+        if (param.isAssignmentPattern()) {
+          body.push(buildLooseDefaultParam({
+            VARIABLE_NAME: param.get("left"),
```

Comment:
Thank you I did not consider that case yet. Since destructuring has default values on its own I only have to handle the top-level default when parameter destructuring is nested right? 

I am currently trying to figure out how to write the `if` condition in loose mode. Would the following be correct semantics?

```javascript
function test({a: { b : c} = {}} = {}) {}
```

Will be transformed into:

```javascript
function test(a) {
  if( a === undefined ) {
    let { a : { b : c }} = {}
  }
}
```